blog/_src/posts/2018-10-06-a-spectrum-of-type-soundness-and-performance.md

```diff
@@ -0,0 +1,131 @@
+different ways of thinking about the integrity of programs that combine 
+statically typed and dynamically typed code.
+Recently, we have been sorting them out.
+
+<!-- more -->
+
+> Note: this post is an extended abstract for the paper _A Spectrum of Type
+  Soundness and Performance_ by Ben Greenman and Matthias Felleisen.
+  For the full paper, slides, code, and a video presentation, visit
+  <http://www.ccs.neu.edu/home/types/publications/publications.html#gf-icfp-2018>
+
+A dynamically-typed language runs any program that "looks good" (i.e.,
+ passes some basic syntactic criteria ... in Python a program cannot mix
+ indentation levels, in Racket a program cannot refer to unbound variables).
+A statically-typed language runs any program that both "looks good" and
+ passes an extra set of type checks.
+
+A _mixed-typed_ language allows some combination of static and dynamic typing.
+There are many languages that fall in the mixed-typed category; figure 1 lists
+ a few, roughly arranged left-to-right by the year they first provided a way to
+ mix.
+
+![Figure 1: Some mixed-typed languages](/img/mixed-typed-systems-by-year.png)
+
+These languages all try to combine static and dynamic typing in a useful way,
+ but they take VERY different approaches.
+For example:
+
+- **MACLISP** defines a syntax for type annotations but does not say how a compiler
+  should interpret the types; see section 14.2 of the [Moonual](http://www.softwarepreservation.org/projects/LISP/MIT/Moon-MACLISP_Reference_Manual-Apr_08_1974.pdf).
+  For example, a compiler may use types to generate specialized code that assumes
+  the type annotations are correct.
+- **Strongtalk** includes a static type checker and DOES NOT use types to change the
+  behavior of a program.
+  For rationale, see the [Pluggable Type Systems](http://bracha.org/pluggableTypesPosition.pdf) position paper.
+- **Typed Racket** lets a program combine statically-typed modules and dynamically-typed
+  modules. The compiler inserts run-time checks at the boundaries between such
+  modules to detect any mismatches between the static types and incoming dynamically-typed
+  values.
+- **Thorn** requires that every value in a program has a type, but allows
+  dynamically-typed contexts to manipulate values. In other words, every Thorn
+  value is an instance of a statically-declared class and classes may contain
+  dynamically-typed methods.
+- **Reticulated** lets a program combine static and dynamic _expressions_ and
+  guarantees that the combination has a well-defined semantics (Vitousek, Swords, and Siek [POPL 2017](https://dl.acm.org/citation.cfm?id=3009849)).
+
+That makes five different systems.
+There are 15 other systems in the figure, and many more in the world.
+One thing we could do is _classify_ them all, but it is doubtful (to me) that a
+ classification would help us understand new systems that appear in the future.
+Instead ...
+
```

Comment:
I don't understand the transition here/what you are saying the motivation is.

Reply:
OK, I think this is better, how about editing the "three ways to define the behavior of programs" line in the next paragraph to refer to the behavior at the boundary?

I'd also expect some mention that these three boundary behaviors are meant to account for *all* of the systems (or not).

Reply:
yep that sounds good (I hadn't commented here yet b/c I was planning to re-read the whole thing first)